core/src/com/unciv/logic/map/tile/TileInfoImprovementFunctions.kt

```diff
@@ -127,9 +127,9 @@ class TileInfoImprovementFunctions(val tile: Tile) {
             improvement.name == Constants.cancelImprovementOrder -> (tile.improvementInProgress != null)
             // Can only remove roads if that road is actually there
             RoadStatus.values().any { it.removeAction == improvement.name } -> tile.roadStatus.removeAction == improvement.name
-            // Can only remove features if that feature is actually there
+            // Can only remove features or improvement if that feature/improvement is actually there
             improvement.name.startsWith(Constants.remove) -> tile.terrainFeatures.any { it == improvement.name.removePrefix(
-                Constants.remove) }
+                Constants.remove) } || tile.improvement == improvement.name.removePrefix(Constants.remove)
```

Comment:
Not sure you can do `val` here and still keep it inline unless you put it earlier in the function. For improvements without "Remove" at the start of the name, not sure it would be better to do so (probably would be worse/premature optimization). If you're suggesting just making it not inline... I guess, but that's not the cleanest thing in the world. Seems like overkill

Interestingly, kotlin checks `startsWith` in `removePrefix`, so technically comparing `Constants.remove + it == improvement.name` and `Constants.remove + tile.improvement == improvement.name` would I assume be in theory faster. Maybe premature optimization, but at least there's a fair argument for it here. Still doing 2 assigns over one if we just used `val`, but again, how much are we really saving here for not being inline?